vllm/v1/core/kv_cache_manager.py

```diff
@@ -471,6 +422,21 @@ def _touch(self, blocks: List[KVCacheBlock]) -> None:
                 self.free_block_queue.remove(block)
             block.incr_ref()
 
+    def _untouch(self, blocks: List[KVCacheBlock]) -> None:
+        """Untouch a block decreases its reference count by 1, and may add
+        the block to the free queue. This is used to reverse touched blocks
+        that are hit by another request with the same prefix.
+
+        Args:
+            blocks: A list of blocks to touch.
+        """
+        for block in blocks:
+            block.decr_ref()
+            # ref_cnt=0 means this block can be added in the free list
+            # (i.e. eviction candidate).
+            if block.ref_cnt == 0:
+                self.free_block_queue.append(block)
```

Comment:
This function breaks the LRU cache policy, because when you touch a block and untouch it, you moved the block to the end of the free queue. It means the block is "used" recently and is less likely evicted. However, In the case that we cannot allocate new blocks for a request, the computed blocks cannot be treat as "recent used".